### ISSUE TITLE
Suggestions stays after input onblur

### ISSUE THREAD

**Issue by jjavierdguezas** [xuopled/react-google-places-suggest#28]
**Steps to reproduce:** go to the [example page](https://xuopled.github.io/react-google-places-suggest/), type something in the input, then click outside the suggestions fragment

**Desired behavior:** Suggestions are hidden after the input loses the focus

_... maybe if you expose the_ [`open`](https://github.com/xuopled/react-google-places-suggest/blob/64c78b2c4a1ab5bc0aed9d93aa7b3c4e6fdcb6b9/src/index.js#L19) _var_

**Comment by xuopled** [comment]
Maybe you can do:
```js
import React, {Component} from "react"
import GoogleMapLoader from "react-google-maps-loader"
import GooglePlacesSuggest from "react-google-places-suggest"

const MY_API_KEY = "AIzaSyDwsdjfskhdbfjsdjbfksiTgnoriOAoUOgsUqOs10J0" // fake

export default class GoogleSuggest extends React.Component {
    state = {
        search: "",
        value: "",
    }

    handleInputChange = e => {
        this.setState({search: e.target.value, value: e.target.value})
    }
    
    handleInputBlur = e => {
        this.setState({search: ""})
    }

    handleSelectSuggest = (geocodedPrediction, originalPrediction) => {
        console.log(geocodedPrediction, originalPrediction) // eslint-disable-line
        this.setState({search: "", value: geocodedPrediction.formatted_address})
    }

    render() {
        const {search, value} = this.state
        return (
            <ReactGoogleMapLoader
                params={{
                    key: MY_API_KEY,
                    libraries: "places,geocode",
                }}
                render={googleMaps =>
                    googleMaps && (
                        <ReactGooglePlacesSuggest
                            googleMaps={googleMaps}
                            autocompletionRequest={search && {
                                input: search,
                                // Optional options
                                // https://developers.google.com/maps/documentation/javascript/reference?hl=fr#AutocompletionRequest
                            }}
                            // Optional props
                            onSelectSuggest={this.handleSelectSuggest}
                            textNoResults="My custom no results text" // null or "" if you want to disable the no results item
                            customRender={prediction => (
                                <div className="customWrapper">
                                    {prediction
                                        ? prediction.description
                                        : "My custom no results text"}
                                </div>
                            )}
                        >
                            <input
                                type="text"
                                value={value}
                                placeholder="Search a location"
                                onChange={this.handleInputChange}
                                onBlur={this.handleInputBlur}
                            />
                        </ReactGooglePlacesSuggest>
                    )
                }
            />
        )
    }
}
```

**Comment by jjavierdguezas** [comment]
Hello @xuopled, first thank you for this library and for helping me with this issue
Did you test your code?
I ask you because I tryed it in my project and, as I expected, that code has a collateral error: when I select a suggestion the `onBlur` event is triggered so the selection does not take place.
If you tested it and it worked, then I'll have to check my code and isolate all the other components and packages to find the problem
Thank you

**Comment by xuopled** [comment]
No sorry I did not test this code. I understand what you mean. I need to think about this. I'm sorry I don't have a lot of free times but I'm very open if you want to make a PR to help me :)

**Comment by jjavierdguezas** [comment]
I'm sorry, I don't have much free time either. It occurs to me right now to delete [this on lines 31-32](https://github.com/xuopled/react-google-places-suggest/blob/64c78b2c4a1ab5bc0aed9d93aa7b3c4e6fdcb6b9/src/index.js#L32) and move [line 55](https://github.com/xuopled/react-google-places-suggest/blob/64c78b2c4a1ab5bc0aed9d93aa7b3c4e6fdcb6b9/src/index.js#L55) to [line 60](https://github.com/xuopled/react-google-places-suggest/blob/64c78b2c4a1ab5bc0aed9d93aa7b3c4e6fdcb6b9/src/index.js#L60) in [index.js](https://github.com/xuopled/react-google-places-suggest/blob/64c78b2c4a1ab5bc0aed9d93aa7b3c4e6fdcb6b9/src/index.js). What do you think? If not, I'll take the time to think it through and make a PR. Maybe on the weekend 😉

**Comment by xuopled** [comment]
It sounds like a good idea

**Comment by jjavierdguezas** [comment]
@xuopled my idea had the same problem that the one that you were proposing. Suggestions are hidden but none can be selected. After some thought and code tests I noticed that similar approaches would not work without making a lot of changes in the implementation of the component. This is because the only way to "communicate" with the component is by modifying the `props` and this causes the `componentWillReceiveProps` to be triggered and prevents the `handleSelectPrediction` method from being executed when clicking on a suggestion.
That makes necessary to detect when a click is made outside the drawn area (the area of the input and the suggestions) and on that instant to hide the suggestions.
The `List` component exactly occupies the drawing area, so there's where I focused. Since the component is a `div` the `onblur` event don't work very well. So in my solution I use a combination of the `onmouseenter` and` onmouseleave` events to determine if the suggestions were "active". Then in the main component I subscribed to the `click` event of `document` and for each click it is verified if it was clicked inside or outside of the suggestions area and hidden if so. Mouse events in the `List` component help to determine this very quickly. Another approach could be [this](http://blustemy.io/detecting-a-click-outside-an-element-in-javascript/).
Well I have done the pull request, I leave it in your hands
